### ISSUE TITLE
override OS name and skip dependencies on install

### ISSUE THREAD

**Issue by akostadinov** [rvm/rvm#3735]
Hello,
I wanted to install rvm in a restrictive environment where a lot of things are not allowed for user, packages cannot be discovered and installed, home dir is not writable.

So I tried 
```
curl -sSL https://get.rvm.io | bash -s stable --path "$WRITABLE_DIR"/rvm
. "$WRITABLE_DIR"/rvm/scripts/rvm
rvm install ruby-2.3.1
```

Distribution is Red Hat Enterprise Linux. It failed for two reasons.

1. no binaries found for "redhat" operating system
1. then (once I worked around /see below)/, detection of dependencies failed due to file system access permissions 

My workaround is to do:
```
  . "$WRITABLE_DIR"/rvm/scripts/rvm
  sed -i "$WRITABLE_DIR"/rvm/scripts/functions/detect_system -e 's/-RedHat/-CentOS/'
  cat >> "$WRITABLE_DIR"/rvm/scripts/functions/requirements/centos <<EOF
  requirements_centos_before_update() { return 0; }
  requirements_centos_libs_install() { return 0; }
  requirements_centos_before_install_epel() { return 0; }
  requirements_centos_before_update_check_epel() { return 0; }
EOF
```

Basically I set CentOS to be default and make the dependency installation bash functions noop.

My proposals:

1. alias `centos` ruby binaries for install on `redhat` (e.g. install centos binaries on both redhat and centos machines)
2. allow user to override operating system name by a command line option
3. allow user to skip dependency checks and installation by a command line option

This issue might be kind of isolated use case but I think these changes can come in handy when one wants to hack something up in a restricted environment without root access. I heard some people have managed machines without root, I assume it could be handy for such situations as well.
